validate.go

```diff
@@ -18,34 +19,127 @@ func validate(payload []byte) ([]byte, error) {
 
 	logger.Info("validating request")
 
-	data := gjson.GetBytes(
+	if len(settings.AllowedHostPaths) == 0 {
+		// empty settings, accepting
+		return kubewarden.AcceptRequest()
+	}
+
+	volumes := gjson.GetBytes(
 		payload,
-		"request.object.metadata.name")
+		"request.object.spec.volumes")
+	if !volumes.Exists() {
+		// pod defines no volumes, accepting
+		return kubewarden.AcceptRequest()
+	}
+
+	var volumeMounts gjson.Result
+	// workaround bug when obtaining array
+	// request.object.spec.containers.volumeMounts
+	for _, object := range gjson.GetBytes(payload,
+		"request.object.spec.containers").Array() {
```

Comment:
As already discussed, this policy has to consider `initContainers` too.

Reply:
Indeed! I totally overlooked that.

Reply:
Done in https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/ead60048e6a00223a677051e4c2026e2f43302c1, with tests in https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/4faa364cdb9d9f522a8ae688a47188976c1e6337 and https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/5333d9074e44da62bbdaabf7817ad8df4f54ebe4.